### DISCUSSION TITLE
How to get and validate the value of an input in taiko?

### DISCUSSION THREAD

**Discussion by Franciscojose05** [getgauge/taiko#2614] [Development]
Hi, i would like to get the value of an input in taiko.
In JS i would do something like this: 
window.document.querySelectorAll('#beitrag_wert')[0].shadowRoot.getElementById('beitrag_wert-control').value

How do i get the value and validate it, whether it is lets say = 50?

**Comment by zabil** [comment]
You can use the [evaluate](https://docs.taiko.dev/running_scripts_inside_the_browser/#perform-actions-on-the-page) to run a script of the browser and return it's value.

for example

```
var value = await evaluate(function() { 
   return window.document.querySelectorAll('#beitrag_wert')[0].shadowRoot.getElementById('beitrag_wert-control').value
});
```

**Reply by Franciscojose05** [reply]
Thx @zabil , that works with a number.
But why I cannot pass a parameter like so?
step("Validate: <i> for <id> should be <expected>", async function (i, id, expected) {
    let dezimalvalue = await evaluate(function() { 
        return window.document.querySelectorAll('#beitrag_wert')[i].shadowRoot.getElementById('beitrag_wert-control').value
     });
     assert.equal(dezimalValue, expected);
});

**Reply by zabil** [reply]
There's a section about passing [parameters](https://docs.taiko.dev/running_scripts_inside_the_browser/#passing-data) which will help solve your problem.
